src/cfnlint/__init__.py

```diff
@@ -595,8 +595,13 @@ def get_directives(self):
         """ Get Directives"""
         results = {}
         for _, resource_values in self.template.get('Resources', {}).items():
-            ignore_rule_ids = resource_values.get('Metadata', {}).get('cfn-lint', {}).get('config', {}).get('ignore_checks', [])
-            for ignore_rule_id in ignore_rule_ids:
+            ignore_rule_ids = set(resource_values.get('Metadata', {}).get('cfn-lint', {}).get('config', {}).get('ignore_checks', []))
+            mandatory_rule_ids = set([])
+            try:
+                mandatory_rule_ids = set(cfnlint.config.ConfigMixIn.mandatory_checks)
+            except:
+                pass
+            for ignore_rule_id in (ignore_rule_ids - mandatory_rule_ids):
```

Comment:
Ignore rules can be the prefix of the string as well.  So you can do E1 and that would ignore all E1XXX rules.  

https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/__init__.py#L206-L207

Reply:
The other thing we may need to figure out here is how to handle the directives section of exceptions.  So people can put in an exception in a resources metadata.  Those exceptions are filtered out here.  In this case the rule still runs but we filter out the results based on these directives.

https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/__init__.py#L1356-L1369